### PR TITLE
assert_labels: report all missing labels

### LINKED ISSUE #480: assert_labels: report *all* missing labels, not just the first to be spotted

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

fixes #480

### CODE REVIEW

**Comment by twaugh** [comment]
Missing unit testing?

File: atomic_reactor/plugins/pre_assert_labels.py [hunk 1]
```diff
@@ -37,8 +37,11 @@ def run(self):
         run the plugin
         """
         labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
+        failed = False
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_assert_labels.py]: What about:
```
missing_labels = filter(lambda x: x if not labels.get(x), self.required_labels)
if missing_labels:
    raise RuntimeError("Missing labels: %s" % missing_labels)
```

File: atomic_reactor/plugins/pre_assert_labels.py [hunk 2]
```diff
@@ -37,8 +37,11 @@ def run(self):
...
         for label in self.required_labels:
             if labels.get(label) is None:
                 msg = "Dockerfile is missing '{0}' label.".format(label)
                 self.log.error(msg)
-                raise AssertionError(msg)
+                failed = True
+        if failed:
+            raise AssertionError('Docker file is missing required labels.')
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_assert_labels.py]: `AssertionError` is rather uncommon, maybe `RuntimeError` as in other plugins?

Comment (rcerven) [inline atomic_reactor/plugins/pre_assert_labels.py]: ok

File: atomic_reactor/plugins/pre_assert_labels.py (hunk 1 above)
Comment (rcerven) [inline atomic_reactor/plugins/pre_assert_labels.py]: changed

**Comment by rcerven** [comment]
unit test added

**Pull request merged**